ContactMechanics/Systems.py

```diff
@@ -659,3 +703,34 @@ def dual_hessian_product(self, pressure):
         res = self.substrate.nb_domain_grid_pts
         hessp = self.substrate.evaluate_disp(-pressure.reshape(res))
         return hessp.reshape(-1)
+
+    def dual_minimize_proxy(self, offset, solver=bugnicourt_cg,
+                            **kwargs):
+        """
+        Convenience function. Eliminates boilerplate code for DUAL minimisation
+        problems by encapsulating the use of constrained minimisation.
+
+        Parameters:
+        offset     -- determines indentation depth
+        disp0      -- initial guess for surface displacement. If not set, zero
+                      displacement of shape
+                      self.substrate.nb_domain_grid_pts is used
+        maxiter    -- maximum number of iterations allowed for convergence
+        logger     -- optional logger, to be used with a logger from
+                      PyCo.Tools.Logger
+        """
+        # pylint: disable=arguments-differ
+        self.disp = None
+        self.force = None
+        self.contact_zone = None
+        result = solver.constrained_conjugate_gradients(
+            self.dual_objective(offset, gradient=True),
+            self.dual_hessian_product,
+            **kwargs)
+        if result.success:
+            self.offset = offset
+            self.disp = result.jac
+            self.force = self.substrate.force = result.x
```

Comment:
@pastewka , is this still fitting in our code style ? I think we abolished caching the energies and forces into the adhesion potential but store it as an attribute of the system. 

I think we should do the same on substrate side for consistency

Reply:
this would need to be changed in the origninal minimize_proxy as well

Reply:
I open an issue for this point.  See Issue #38